### PR TITLE
Remove old transfer_dev_to_mem and update ad7134 to use iio_app

### PR DESCRIPTION
**Pull request by mchindri**
Remove old transfer_dev_to_mem function and use new one. Then remove it from iio.
Add support for register access.
Use iio_app

### CODE REVIEW

**Comment by mchindri** [comment]
v2:
 - Fix CI problem

**Review by dbogdan — APPROVED** [review]

**Comment by dbogdan** [comment]
By the way, was this tested on the real hardware?

**Comment by mchindri** [comment]
> By the way, was this tested on the real hardware?

Yes, but I have some issue 

> By the way, was this tested on the real hardware?

Yes, but there is still a issue here:  https://github.com/analogdevicesinc/no-OS/blob/a26ec68e988a748973d30fd2f6850b4e9913d80d/drivers/adc/ad713x/iio_dual_ad713x.c#L123
It has different behavior than https://github.com/analogdevicesinc/no-OS/blob/a26ec68e988a748973d30fd2f6850b4e9913d80d/projects/ad713x_fmcz/src/ad713x_fmc.c#L277
I asked @cristiantpop to take a look at the PR maybe I am missing something.

File: iio/iio_ad9361/iio_ad9361.c [hunk 1]
```diff
@@ -2296,10 +2296,6 @@ int32_t iio_ad9361_init(struct iio_ad9361_desc **desc,
 	iio_ad9361_inst->dev_descriptor.attributes = global_attributes;
 	iio_ad9361_inst->dev_descriptor.debug_attributes = NULL;
 	iio_ad9361_inst->dev_descriptor.buffer_attributes = NULL;
-	iio_ad9361_inst->dev_descriptor.transfer_dev_to_mem = NULL;
```
Comment (amiclaus) [inline iio/iio_ad9361/iio_ad9361.c:2299]: It looks that you are doing here changes also to other projects that are unrelated to ad7134.

File: libraries/iio/iio.c [hunk 2]
```diff
@@ -819,12 +819,6 @@ static uint32_t bytes_to_samples(struct iio_interface *intf, uint32_t bytes)
 static ssize_t iio_transfer_dev_to_mem(const char *device, size_t bytes_count)
 {
 	struct iio_interface *iio_interface = iio_get_interface(device);
-
-	if (iio_interface->dev_descriptor->transfer_dev_to_mem)
```
Comment (amiclaus) [inline libraries/iio/iio.c:823]: Shouldn't library changes be subject for another PR? Since these changes might influence also other projects than ad7134.

File: iio/iio_app/iio_app.c [hunk 3]
```diff
@@ -36,6 +36,9 @@
  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *******************************************************************************/
+
+#ifdef IIO_SUPPORT
```
Comment (amiclaus) [inline iio/iio_app/iio_app.c:40]: another change that has to do with more than ad7134.

**Comment by mchindri** [comment]
v3: 
 - Fix _iio_ad713x_read_dev. No it works fine. (tested on hardware)

File: libraries/iio/iio.c (hunk 2 above)
Comment (mchindri) [reply libraries/iio/iio.c:823]: Only this projects was still using this functions. The scope was to remove them so the project has to be updated. 
Anyway, the changes are in separated commits.

File: iio/iio_app/iio_app.c (hunk 3 above)
Comment (mchindri) [reply iio/iio_app/iio_app.c:40]: it is a commit unrelated to ad7134.

Comment (amiclaus) [reply iio/iio_app/iio_app.c:40]: I was a bit confused about the PR title.

Comment (mchindri) [reply iio/iio_app/iio_app.c:40]: My mistake, I changed it :)

**Review by amiclaus — APPROVED** [review]
